presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -1005,6 +1007,28 @@ public void testUnnest()
     }
 
     @Test
+    public void testLambda()
+            throws Exception
+    {
+        assertStatement("SELECT x -> sin(x) FROM t",
```

Comment:
Interestingly, that wouldn't be a valid query :)

Use `assertExpression`, instead.